src/stores/rootStore.ts

```diff
@@ -0,0 +1,389 @@
+    },
+
+    /** Is True if this is a Paid or Completed Application. */
+    isAppFiling (state): boolean {
+      return (state.entityStatus === EntityStatus.FILED_APP)
+    },
+
+    /** The Name Request (may be null). */
+    getNameRequest (state): any {
+      return (state.nameRequest)
+    },
+
+    isCurrentFilingEditable (state): boolean {
+      return (state.currentFilingStatus === FilingStatus.NEW || state.currentFilingStatus === FilingStatus.DRAFT)
+    },
+
+    getReportState (state): string {
+      switch (state.currentFilingStatus) {
+        case FilingStatus.NEW: return ''
+        case FilingStatus.DRAFT: return 'Draft'
+      }
+      return state.currentFilingStatus
+    },
+
+    /** Whether to show the Fetching Data spinner. */
+    showFetchingDataSpinner (state): boolean {
+      return state.fetchingDataSpinner
+    },
+
+    /**
+     * This is used to show Legal Obligations only for a new business
+     * that has no tasks and hasn't filed anything yet (except their IA).
+     **/
+    isBusinessWithNoMaintenanceFilings (state): boolean {
+      const filingHistoryListStore = useFilingHistoryListStore()
+      return (
+      // no todo items
+        (state.tasks.length === 0) &&
+      // only the IA or Registraion filing history item
+      (filingHistoryListStore.getFilings.length === 1) && (
+          EnumUtilities.isTypeIncorporationApplication(filingHistoryListStore.getFilings[0]) ||
+        EnumUtilities.isTypeRegistration(filingHistoryListStore.getFilings[0])
+        )
+      )
+    },
+
+    /** The entity registered office address. */
+    getRegisteredOfficeAddress (state): OfficeAddressIF {
+      return state.registeredAddress
+    },
+
+    /** The parties list from the API. */
+    getParties (state): Array<PartyIF> {
+      return state.parties
+    },
+
+    /** The entity business address. */
+    getBusinessAddress (state): OfficeAddressIF {
+      return state.businessAddress
+    },
+
+    /** The entity resource text for confirmation modal. */
+    getDissolutionConfirmationResource (state): DissolutionConfirmationResourceIF {
+      return state.configObject?.dissolutionConfirmation
+    },
+    /** The entity TodoList resources. */
+    getTodoListResource (state): TodoListResourceIF {
+      return state.configObject?.todoList
+    },
+
+    /** The corp type code from Auth db (may be null). */
+    getCorpTypeCd (state): CorpTypeCd {
+      return state.corpTypeCd
+    },
+
+    //
+    // State Filing getters
+    //
+
+    /** The filing that changed the business state, if there is one. */
+    getStateFiling (state): StateFilingIF {
+      return state.stateFiling
+    },
+
+    /**
+     * A formatted concatenation of the name and the effective date of the filing.
+     * Only used when entity is historical.
+     */
+    getReasonText (): string {
+      const businessStore = useBusinessStore()
+      const stateFiling = this.getStateFiling as StateFilingIF // may be null
```

Comment:
Is the "as" needed or does getStateFiling already have a type?

Reply:
It's the same scenario as the `ApiFilingIF[]` cast above. If we do not use the as, my IDE is saying that stateFiling has a type of any if I hover over it.

Reply:
OK, leave it.